From 611e93f2f8c0c18ed8d365ecb850258e8d5c9c12 Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@gnome.org>
Date: Wed, 16 Jan 2013 22:52:03 +0100
Subject: [PATCH] window: Use the geometry of the active screen

When we get the restyle on focus-out, the screens call terminal_window_set_size()
directly. This was causing the window to revert to a previous size, see bug
https://bugzilla.gnome.org/show_bug.cgi?id=688959 .
---
 src/terminal-window.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/terminal-window.c b/src/terminal-window.c
index ed509ba..1740247 100644
--- a/src/terminal-window.c
+++ b/src/terminal-window.c
@@ -2330,6 +2330,10 @@ terminal_window_set_size_force_grid (TerminalWindow *window,
   int grid_width;
   int grid_height;
 
+  /* Only update the geometry from the active screen. */
+  if (window->priv->active_screen != screen)
+    return;
+
   /* be sure our geometry is up-to-date */
   terminal_window_update_geometry (window);
 
-- 
1.8.1.1

